core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthChecker.java

```diff
@@ -105,8 +106,37 @@ static HealthChecker ofCpu(double targetSystemCpuUsage, double targetProcessCpuU
         return new CpuHealthChecker(targetSystemCpuUsage, targetProcessCpuUsage);
     }
 
+    /**
+     * Creates a new instance of {@link HealthChecker} which reports health based on
+     * cpu usage reported by {@link OperatingSystemMXBean}.
+     * Both system and process cpu usage must be (inclusive) lower than the specified threshold in order
+     * for the {@link HealthChecker} to report as healthy. If the {@link HealthChecker} is unable
+     * to find a suitable {@link OperatingSystemMXBean}, an exception is thrown on construction.
+     *
+     * @param targetSystemCpuUsage Target system CPU usage as a percentage (0 - 1).
+     * @param targetProcessCpuUsage Target process CPU usage as a percentage (0 - 1).
+     * @param degradedTargetSystemCpuUsage Degraded target system CPU usage as a percentage
+     *     (targetSystemCpuUsage - 1).
+     * @param degradedTargetProcessCpuLoad Degraded target process CPU usage as a percentage
+     *     (targetProcessCpuUsage - 1).
+     * @return an instance of {@link HealthChecker} configured with the provided CPU usage targets.
+     */
+    static HealthChecker ofCpu(double targetSystemCpuUsage, double targetProcessCpuUsage,
+                               double degradedTargetSystemCpuUsage, double degradedTargetProcessCpuLoad) {
+        return new CpuHealthChecker(targetSystemCpuUsage, targetProcessCpuUsage,
+                                    degradedTargetSystemCpuUsage, degradedTargetProcessCpuLoad);
+    }
+
     /**
      * Returns {@code true} if and only if the {@link Server} is healthy.
      */
     boolean isHealthy();
```

Comment:
Note: I imagine that some time in the future we may likely end up deprecating this API since we're only using `HealthChecker#healthStatus` internally in `HealthCheckService`.

I prefer this be handled separately though if necessary.

Reply:
Meanwhile, can you go through the code in armeria and convert any usages using this API to use `healthStatus` instead??

Reply:
I've updated the code to use `healthStatus()` instead of using `isHealthy()` API. The only place where `isHealthy()` is used is below method. 

```java 
default HealthStatus healthStatus() {
    return isHealthy() ? HealthStatus.HEALTHY : HealthStatus.UNHEALTHY;
}
```